solution/index.js

```diff
@@ -0,0 +1,170 @@
+// Add new Task 
+function handleAddTask (event){
+    let newTask = event.target.nextElementSibling.value;
+    if(newTask === ""){
+        alert("can't add empty task");
+        return;
+    }
+    let newTaskEl = document.createElement("li");
+    newTaskEl.classList.add("task");
+    newTaskEl.innerText = newTask;
+    newTaskEl.contentEditable = false;
+    let taskList = document.getElementsByTagName("ul")[event.target.parentNode.id];
+    taskList.insertBefore(newTaskEl, taskList.childNodes[0]);
+    tasks[event.target.parentNode.dataset.action].unshift(newTask);  
+    localStorage.setItem("tasks", JSON.stringify(tasks));  //save to local storage
+    
+}
+//Edit Task
+function handleDBclick (event){
+    if(event.target.tagName !== "LI") return;
```

Comment:
you added the event to the document level so it will run more times than you need. yes you created an if to handle the problem but thats not the correct way to handle this task.